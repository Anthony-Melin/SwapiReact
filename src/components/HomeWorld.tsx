import { PropsWithChildren } from 'react'

type HomeWorldProps = PropsWithChildren<{ value: string }>

const HomeWorld = ({ value, ...restProps }: HomeWorldProps) => {
    return (
        <p datatest-id="homeworld" {...restProps}>
            HomeWorld: {value}
        </p>
    )
}

export default HomeWorld
