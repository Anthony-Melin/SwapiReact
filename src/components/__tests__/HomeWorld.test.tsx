import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import HomeWorld from '../HomeWorld.tsx'

describe('HomeWorld', () => {
    beforeEach(() => {
        render(<HomeWorld value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('homeworld')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('homeworld')).toHaveTextContent(
            'Home world: VALUE'
        )
    })
})
