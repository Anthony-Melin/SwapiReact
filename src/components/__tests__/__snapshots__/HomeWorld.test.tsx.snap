// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeWorld > should match snapshot 1`] = `
<p
  data-testid="homeworld"
>
  Home world
  : 
  VALUE
</p>
`;
